### PR TITLE
fix: handle long extracted text and search occurences

### PR DESCRIPTION
**Pull request by caro3801**
Related to the bug: ICIJ/datashare#1033
And related to modifications here: https://github.com/ICIJ/datashare/pull/1043

Fixed:
- Replace virtual scroller by paginated extracted text

Added:
- Extracted text download as text file is now available under Download document

Improvement suggestions:
- Refactor document content component (DRY)
- Prevent download or display when extracted text is empty.
- Propose a better pagination UX
- Propose a better download extracted text UX

Misc: 
- when merging, I'll try to squash the PR in 2 commits: 1 for the "fixed", 1 for the "added"

### CODE REVIEW

File: src/components/DocumentTranslatedContent.vue [hunk 1]
```diff
@@ -97,7 +97,7 @@ export default {
   },
   methods: {
     toggleTranslatedContent() {
-      this.$store.commit('document/toogleShowTransatedContent')
+      this.$store.commit('document/toggleShowTranslatedContent')
```
Comment (pirhoo) [inline src/components/DocumentTranslatedContent.vue:100]: We could say this is how I sign every code I produce :upside_down_face:

File: src/components/document/DocumentNavbar.vue [hunk 2]
```diff
@@ -95,7 +95,7 @@ import utils from '@/mixins/utils'
  * Document navbar in the context of a search.
  */
 export default {
-  name: 'SearchDocumentNavbar',
+  name: 'DocumentNavbar',
```
Comment (pirhoo) [inline src/components/document/DocumentNavbar.vue:98]: Thanks for catching this! I'm surprised it never had any side effects.

**Review by pirhoo — APPROVED** [review]
Brilliant, thanks!

File: src/components/DocumentContent.vue [hunk 3]
```diff
@@ -374,17 +410,27 @@ export default {
...
+          v-model="page"
+          align="center"
+          class="m-0"
+          size="sm"
+          :per-page="1"
+          :total-rows="nbPages"
+        ></b-pagination>
+      </div>
+      <b-overlay :show="$wait.is('loadContentSlice')" opacity="0.6" rounded spinner-small>
+        <hook name="document.content.body:before"></hook>
```
Comment (caro3801) [inline src/components/DocumentContent.vue:425]: @pirhoo Should this hook `document.content.body:before` go before the pagination ? (L412) (same question for `document.content.body:after`

Comment (pirhoo) [reply src/components/DocumentContent.vue:425]: Good question: this hook might be used to add elements "around" the pagination block so I'm tempted to move it just outside of the `b-overlay`

**Review by mvanzalu — APPROVED** [review]

**Comment by caro3801** [comment]
Due to hard code maintenance and bugs still happening on the search, the organic cooking of slicing has been **removed**
Added: Highlight of cut words (search on 2 pages)

**Pull request merged**